### PR TITLE
Adds Position, Order, and Portfolio History Models

### LINKED ISSUE #55: Trading API: Portfolio History
Implement routes defined by Trading API [portfolio history docs](https://alpaca.markets/docs/api-references/trading-api/portfolio-history/). 

```python

def get_portfolio_history(self, request: GetPortfolioHistoryRequest) -> PortfolioHistory:
   pass

```

### PR DESCRIPTION
**Pull request by ryansdowning**
Partially resolves #55 and #56.

### CODE REVIEW

File: tests/common/test_common_models.py [hunk 1]
```diff
@@ -27,3 +37,240 @@ def test_calendar_timestamps():
...
+        trail_price=None,
+        hwm=None,
+        commission="1.25",
+    )
+
+    assert isinstance(order_with_legs.legs, list)
+    assert isinstance(order_with_legs.legs[0], Order)
+
+
+class TestClosePositionRequest:
```
Comment (drew887) [inline tests/common/test_common_models.py:249]: Why did you make this a test class? If possible can we please keep the tests as free functions ?

**Review by drew887 — CHANGES_REQUESTED** [review]

File: tests/common/test_common_models.py (hunk 1 above)
Comment (ryansdowning) [reply tests/common/test_common_models.py]: Fixed.

**Comment by ryansdowning** [comment]
Unfortunately I do not have access to the sonarcloud results so I am unable to review what is wrong with the code analysis.

File: alpaca/common/models/trading.py [hunk 2]
```diff
@@ -1,28 +1,174 @@
...
+        filled_at (Optional[datetime]): Timestamp when the Order was filled.
+        expired_at (Optional[datetime]): Timestamp when the Order expired at.
+        canceled_at (Optional[datetime]): Timestamp when the Order was canceled.
+        failed_at (Optional[datetime]): Timestamp when the Order failed at.
+        replaced_at (Optional[datetime]): Timestamp when the Order was replaced by a new Order.
+        replaced_by (Optional[UUID]): ID of Order that replaces this Order.
+        replaces (Optional[UUID]): ID of Order which this Order replaces.
+        asset_id (UUID): ID of the asset.
+        symbol (str): Symbol of the asset.
+        asset_class (str): Asset class of the asset.
```
Comment (haxdds) [inline alpaca/common/models/trading.py:76]: `AssetClass`, `OrderClass`, `OrderType`, `OrderSide`, `TimeInForce` should preferably be enums.

File: alpaca/common/models/trading.py [hunk 3]
```diff
@@ -1,28 +1,174 @@
...
+
+    timestamp: List[int]
+    equity: List[float]
+    profit_loss: List[float]
+    profit_loss_pct: List[float]
+    base_value: float
+    timeframe: str
+
+
+class Asset(BaseModel):
```
Comment (haxdds) [inline alpaca/common/models/trading.py:154]: We've already added `Asset` in #61

**Review by haxdds — CHANGES_REQUESTED** [review]
A few things

**Comment by ryansdowning** [comment]
Uses enums from master changes (didn't realize they were added my bad!) and also added a `PositionSide` enum. `Asset` is not re-added in this PR, just a weird commit diff (tried to fix it, but not sure if it worked).

File: alpaca/common/models/trading.py [hunk 4]
```diff
@@ -1,28 +1,184 @@
...
+    filled_at: Optional[datetime]
+    expired_at: Optional[datetime]
+    canceled_at: Optional[datetime]
+    failed_at: Optional[datetime]
+    replaced_at: Optional[datetime]
+    replaced_by: Optional[UUID]
+    replaces: Optional[UUID]
+    asset_id: UUID
+    symbol: str
+    asset_class: str
```
Comment (haxdds) [inline alpaca/common/models/trading.py:147]: Should be of `AssetClass` Type

File: alpaca/common/models/trading.py [hunk 5]
```diff
@@ -1,28 +1,184 @@
...
+    time_in_force: TimeInForce
+    limit_price: Optional[str]
+    stop_price: Optional[str]
+    status: OrderStatus
+    extended_hours: bool
+    legs: Optional[List["Order"]]
+    trail_percent: Optional[str]
+    trail_price: Optional[str]
+    hwm: Optional[str]
+    commission: float
```
Comment (haxdds) [inline alpaca/common/models/trading.py:165]: Let's leave out commission. Broker will have its own Order object extended from this one that can implement commission and any other broker related fields.

**Review by haxdds — CHANGES_REQUESTED** [review]

File: tests/common/test_common_models.py [hunk 6]
```diff
@@ -27,3 +45,139 @@ def test_calendar_timestamps():
...
+
+def test_close_position_request_with_percentage():
+    close_position_request = ClosePositionRequest(percentage="0.5")
+    assert close_position_request.qty is None
+    assert close_position_request.percentage == "0.5"
+
+
+def test_close_position_request_with_qty_and_percentage():
+    with pytest.raises(ValueError) as e:
+        close_position_request = ClosePositionRequest(qty="100", percentage="0.5")
```
Comment (haxdds) [inline tests/common/test_common_models.py:168]: defining a variable `close_position_request` is unnecessary here

File: tests/common/test_common_models.py [hunk 7]
```diff
@@ -27,3 +45,139 @@ def test_calendar_timestamps():
...
+
+    assert (
+        "Only one of qty or percentage must be given to the ClosePositionRequest, got both."
+        in str(e.value)
+    )
+
+
+def test_close_position_request_with_neither_qty_or_percentage():
+    with pytest.raises(ValueError) as e:
+        close_position_request = ClosePositionRequest()
```
Comment (haxdds) [inline tests/common/test_common_models.py:178]: Don't need to define `close_position_request` here either.

**Review by haxdds — CHANGES_REQUESTED** [review]

**Comment by haxdds** [comment]
> Unfortunately I do not have access to the sonarcloud results so I am unable to review what is wrong with the code analysis.

Sonar is complaining about some duplicated lines of code you're using to define the orders in your tests. I market it as false positives.

**Review by haxdds — APPROVED** [review]
looks good, thanks @ryansdowning!

**Review by drew887 — APPROVED** [review]
We're missing some docstrings but in interest of getting this merged I think we can add em later

**Pull request merged**